hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/MessagePublisher.java

```diff
@@ -22,9 +23,11 @@ public MessagePublisher(BrokerMessageProducer brokerMessageProducer) {
         this.brokerMessageProducer = brokerMessageProducer;
     }
 
-    public void publish(Message message, Topic topic, MessageState messageState, BrokerListeners listeners, PublishingCallback callback) {
-        messageState.setState(SENDING_TO_KAFKA_PRODUCER_QUEUE);
-        brokerMessageProducer.send(message, topic, callback);
-        messageState.setState(SENDING_TO_KAFKA);
+    public void publish(Message message, Topic topic, MessageState messageState, RequestTimeoutLock requestTimeoutLock, BrokerListeners listeners, PublishingCallback callback) {
+        if (requestTimeoutLock.tryLock()) {
+            messageState.setState(SENDING_TO_KAFKA_PRODUCER_QUEUE);
+            brokerMessageProducer.send(message, topic, callback);
```

Comment:
if send() throws an exception (for example could not fetch topic metadata) after 65ms then client receives 202 but message will not be send to kafka, only persisted in backup storage.